templates/prescription/general_edit.html

```diff
@@ -305,21 +305,65 @@
     {/literal}
     {if $GLOBALS.weno_rx_enable}
         {literal}
-        $('#drug').autocomplete({
-            {/literal}
-            source: 'library/ajax/drug_autocomplete/search.php?csrf_token_form=' + {$CSRF_TOKEN_FORM|js_url},
-            minLength: 3
-            {literal}
-        });
+        $("#drug").select2({
+        ajax: {
+            url: "library/ajax/drug_autocomplete/search.php",
+            dataType: 'json',
+            data: function(params) {
+                return {
+                  {/literal}
+                  csrf_token_form: {$CSRF_TOKEN_FORM|js_url},
+                  term: params.term
+                  {literal}
+                };
+            },
+            processResults: function(data) {
+                return  {
+                    results: $.map(data, function(item, index) {
+                      return {
+                            text: item,
+                            id: index,
+                            value: item
+                        }
+                    })
+                };
+                return x;
+            },
+            cache: true,
+            minimumInputLength: 3
+            }
+          });
         {/literal}
     {else}
         {literal}
-        $('#drug').autocomplete({
-            {/literal}
-            source: 'library/ajax/prescription_drugname_lookup.php?csrf_token_form=' + {$CSRF_TOKEN_FORM|js_url},
-            minLength: 1
-            {literal}
-        });
+        $("#drug").select2({
+        ajax: {
+            url: "library/ajax/prescription_drugname_lookup.php",
+            dataType: 'json',
+            data: function(params) {
+                return {
+                  {/literal}
+                  csrf_token_form: {$CSRF_TOKEN_FORM|js_escape},
+                  {literal}
+                  term: params.term
+                };
+            },
+            processResults: function(data) {
+                return  {
+                    results: $.map(data, function(item, index) {
+                        return {
+                            text: item,
+                            id: index,
+                            value: item
+                        }
+                    })
+                };
+                return x;
+            },
+            cache: true
+            },
```

Comment:
Gotta add below here to allow free text entering of drug:
```
tags: true,
```
I'm gonna add this in a subsequent commit since this PR is otherwise testing very nicely